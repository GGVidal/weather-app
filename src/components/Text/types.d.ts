import { ReactNode } from "react";
import { TextProps as RnTextProps } from "react-native";

export interface TextProps extends RnTextProps {
  fontSize: string;
  color: string;
  fontWeight: string;
  children: ReactNode;
  lineHeight?: string;
}
