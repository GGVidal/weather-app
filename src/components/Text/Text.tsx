import React, { FC } from "react";
import { StyledText } from "./style";
import { TextProps } from "./types";

export const Text: FC<TextProps> = ({
  color,
  fontSize,
  fontWeight,
  letterSpacing,
  children,
  lineHeight,
}) => {
  return (
    <StyledText
      color={color}
      fontSize={fontSize}
      fontWeight={fontWeight}
      letterSpacing={letterSpacing}
      lineHeight={lineHeight}
    >
      {children}
    </StyledText>
  );
};
