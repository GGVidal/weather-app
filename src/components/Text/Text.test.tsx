import React from "react";
import TestRenderer from "react-test-renderer";

import { Text } from "../Text";

describe("renders correctly", () => {
  it("works", () => {
    const renderer = TestRenderer.create(
      <Text fontSize={"10px"} color={"red"} fontWeight={"300"}>
        My text
      </Text>
    );
    expect(renderer).toMatchSnapshot();
  });
});
