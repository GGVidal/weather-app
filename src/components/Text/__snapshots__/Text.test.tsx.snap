// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly works 1`] = `
<Text
  color="red"
  fontSize="10px"
  fontWeight="300"
  style={
    Array [
      Object {
        "color": "red",
        "fontSize": 10,
        "fontWeight": "300",
        "letterSpacing": "",
        "lineHeight": "",
      },
    ]
  }
>
  My text
</Text>
`;
